src/objectcontroller.cpp

```diff
@@ -1478,6 +1484,299 @@ void DateObject::getDate(int *day, int *month, int *year)
         *year = 1900;
 }
 
+DateTimeObjectValue::DateTimeObjectValue(const std::string& value) : 
+  day_m(-1), month_m(-1), year_m(-1), hour_m(-1), min_m(-1), sec_m(-1)
+{
+    if (value == "now")
+        return;
+    std::istringstream val(value);
+    char s1, s2, s3, s4, s5;
+    val >> year_m >> s1 >> month_m >> s2 >> day_m >> s3 >> hour_m >> s4 >> min_m >> s5 >> sec_m;
+    wday_m = 0;
+    year_m -= 1900;
+    if ( val.fail() ||
+         val.peek() != std::char_traits<char>::eof() || // workaround for wrong val.eof() flag in uClibc++
+         s1 != '-' || s2 != '-' || s3 != ' ' || s4 != ':' || s5 != ':' ||
+         year_m < 0 || year_m > 255 || month_m < 1 || month_m > 12 || day_m < 1 || day_m > 31 ||
```

Comment:
`month_m < 1 || month_m > 12`
Strictly speaking, this is not correct as we copied the entire byte in `doWrite()` without applying a mask to ignore the four non significant trailing bits. But I see the other object types don't seem to do any better than that, so be it.

![image](https://user-images.githubusercontent.com/7327649/117206591-559ddf80-adf3-11eb-9148-a4d30836cb50.png)

Reply:
Thinking about it again, since this is a new class, I would prefer to have it properly implemented. I suggest to apply a mask to consider the relevant bits only. This is applicable to month, day, minutes and seconds:
![image](https://user-images.githubusercontent.com/7327649/117358437-71b68500-aeb6-11eb-94a8-6d7d187b4dad.png)